### PR TITLE
Merge branch `b0.20.1` into develop

### PR DESCRIPTION
**Pull request by tworec**

### CODE REVIEW

File: tests/golem/task/test_taskserver.py [hunk 1]
```diff
@@ -354,7 +354,7 @@ def test_should_accept_provider_no_such_task(self, *_args):
 
         with self.assertLogs(logger, level='INFO') as cm:
             assert not ts.should_accept_provider(
-                node_id, "127.0.0.1", 'tid', 27.18, 1, 1)
+                node_id, "127.0.0.1", 'tid', 27.18, 1, b'')
```
Comment (tworec) [inline tests/golem/task/test_taskserver.py]:
```suggestion
                node_id, "127.0.0.1", 'tid', 27.18, 1, 'oh')
```

**Review by shadeofblue — APPROVED** [review]
didn't read ;)

**Review by mfranciszkiewicz — COMMENTED** [review]
I'm reluctant to approve since these major changes haven't been tested by QA.

Do the integration tests pass?

**Comment by tworec** [comment]
> I'm reluctant to approve since these major changes haven't been tested by QA.
> 
> Do the integration tests pass?

yup! All of them.
Confirmed here.
https://github.com/golemfactory/golem/pull/4440#discussion_r302485115

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**